### PR TITLE
[fix](delete) Fix parsing error of delete where date statement

### LINKED ISSUE #22679: [Bug] Error in parsing the delete statement
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

2.0

### What's Wrong?
For the stock table containing the `date` type:
```
MySQL [testdb]> delete from cost2 where date = '2021-01-01';
ERROR 1105 (HY000): errCode = 2, detailMessage = errCode = 2, detailMessage = Left expr of binary predicate should be column name, predicate=`date` = '2021-01-01 00:00:00'
```

### What You Expected?

Successfully deleted

### How to Reproduce?

```
 ADMIN SET FRONTEND CONFIG ("enable_date_conversion" = "false");

 ADMIN SET FRONTEND CONFIG ("disable_datev1" = "false");

CREATE TABLE `cost` (
  `id` bigint(20) NULL,
  `name` varchar(20) NULL,
  `date` date NULL,
  `cost` bigint(20) SUM NULL
) ENGINE=OLAP
AGGREGATE KEY(`id`, `name`, `date`)
COMMENT 'OLAP'
DISTRIBUTED BY HASH(`id`, `name`) BUCKETS 8
PROPERTIES (
"replication_allocation" = "tag.location.default: 1"
);

insert into cost values (1, 'abc', '2021-01-01', 100);

delete from cost where date = '2021-01-01';
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
## Proposed changes

Issue Number: close #22679

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by luozenglin** [comment]
run buildall

**Comment by luozenglin** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.51 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162185651 Bytes

File: regression-test/suites/delete_p0/test_delete.groovy [hunk 1]
```diff
@@ -250,4 +250,37 @@ suite("test_delete") {
...
+
+    sql "insert into cost values (1, 'abc', '2021-01-01', 100), (2, 'abc', '2021-08-01', 100);"
+
+    qt_delete_date1 "select * from cost order by id"
+
+    sql "delete from cost where date = '2021-01-01';"
+
+    qt_delete_date2 "select * from cost order by id"
+
+    sql 'ADMIN SET FRONTEND CONFIG ("enable_date_conversion" = "true");'
```
Comment (Gabriel39) [inline regression-test/suites/delete_p0/test_delete.groovy:283]: This will cause unpredicatable problems when running regression tests in parallel.

**Review by Gabriel39 — CHANGES_REQUESTED** [review]

**Comment by luozenglin** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.48 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162485965 Bytes

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Review by chenlinzhong — APPROVED** [review]
LGTM

**Pull request merged**